### DISCUSSION TITLE
Relationships filter with more than one filter

### DISCUSSION THREAD

**Discussion by ashgibson** [spatie/laravel-query-builder#617] [Q&A]
In an example of Posts > Comments where I want to find all posts that have comments after a particular date and by one or more specified users, what is the best way to do this?

At the moment i am doing something like this where I pass a nested filter

/users?filter[post_id]=1,2,3&filter[comments][date]=2021-05-17&filter[comments][user_id]=1,2

```
AllowedFilter::exact('post_id'),
AllowedFilter::callback('comments', function (Builder $query, $value) {
                $query->whereHas('comments', function (Builder $query) use ($value) {
                    if ($filter = data_get($value, 'date')) {
                        $query->whereDate('date_time', '>=', $filter);
                    }

                    if ($filter = data_get($value, 'user_id')) {
                        $query->where('user_id', explode(',',$filter));
                    }
                });
            })
```

The problem is that I lose all the nice functionality that this package gives me e.g. the explode() for single/array values.

Anyone have any hints or tips or could the package be improved to handle this?

**Comment by AlexVanderbist** [comment]
Hi @ashgibson! I'd suggest using custom filters for this. One for `comment_date` and one for `comment_user_id`. This way the logic is nice and contained in two separate classes and the package will already take care of parsing the filter value to an array. For a rough example, take a look at the `is_array` section of the `FiltersExact` default filter: https://github.com/spatie/laravel-query-builder/blob/master/src/Filters/FiltersExact.php#L32-L36

Good luck!
